editor/gui/scene_tree_editor.cpp

```diff
@@ -1684,13 +1684,19 @@ void SceneTreeDialog::_show_all_nodes_changed(bool p_button_pressed) {
 }
 
 void SceneTreeDialog::set_valid_types(const Vector<StringName> &p_valid) {
-	if (p_valid.is_empty()) {
-		return;
+	if (allowed_types_hbox) {
+		allowed_types_hbox->queue_free();
```

Comment:
```suggestion
		allowed_types_hbox->queue_free();
		allowed_types_hbox = nullptr;
```
I'm surprised it doesn't crash 🤔

Reply:
I've added the changes from the recent batch of comments.

This one appears to be the solution to the mysterious errors I couldn't figure out earlier. 🎉 

This fix immediately fixed the random message I mentioned:
`core/object/object.h:1036 - Condition "slot >= slot_max" is true. Returning: nullptr`

Also notably, the code used to be
```
content->remove_child(allowed_types_hbox);
allowed_types_hbox->queue_free();
```
And the remove_child actually did crash occasionally. (when quickly toggling between options) I figured remove_child was redundant with the queue_free after so I took it out. 

remove_child seems to work fine with the nullptr added. I hear mixed things about whether a remove_child() is needed before a queue_free. It seems redundant but I see it everywhere in the codebase. Do you think it's worth adding it back?

Reply:
_everywhere_ was probably an exaggeration. :P I did see it both times queue_free was called in this file.